### PR TITLE
Nuke unneeded ClaimsPrincipal extensions

### PR DESCRIPTION
**Pull request by HaoK**
Also fix a bug with custom user id claim type with SecurityStampValidator

cc @divega

### CODE REVIEW

**Comment by HaoK** [comment]
Addresses: https://github.com/aspnet/Identity/issues/663

File: src/Microsoft.AspNet.Identity/PrincipalExtensions.cs [hunk 1]
```diff
@@ -12,36 +12,6 @@ namespace System.Security.Claims
...
-            {
-                throw new ArgumentNullException(nameof(principal));
-            }
-            return principal.FindFirstValue(ClaimTypes.NameIdentifier);
-        }
-
-        /// <summary>
         /// Returns true if the principal has an identity with the application cookie identity
         /// </summary>
         /// <param name="principal">The <see cref="ClaimsPrincipal"/> instance this method extends.</param>
```
Comment (divega) [inline src/Microsoft.AspNet.Identity/PrincipalExtensions.cs]: While you are here, could you fix the null-conditional operator usage in the method bellow? E.g. principal is never null after the check. Or is this the method that is going away per #630?

**Comment by divega** [comment]
:shipit:

File: src/Microsoft.AspNet.Identity/PrincipalExtensions.cs (hunk 1 above)
Comment (HaoK) [inline src/Microsoft.AspNet.Identity/PrincipalExtensions.cs]: Sure, done

**Comment by HaoK** [comment]
1c26e8f5fa1b36febaca1f9a63b2a2f65fb11727

**Comment by HaoK** [comment]
Premature removal, looks like we need to fix templates before we nuke these, as they still rely on this broken sugar method.

**Comment by HaoK** [comment]
Replaced by https://github.com/aspnet/Identity/pull/701

**Pull request closed without merging**